src/arch/xtensa/gdb/debugexception.S

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) 2018, Intel Corporation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the name of the Intel Corporation nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Marcin Rajwa <marcin.rajwa@linux.intel.com>
+ *
+ * Debug context switch.
+ *
+ */
+
+#include <arch/gdb/xtensa-defs.h>
+#define BREAK_INS_IDENTIFIER 0x40
+#define BREAK_INS_SIZE 0x03
+#define PS_EXCM_MASK 0xFFFFFFEF
+#define PS_EXCM_EXCEPTION_MODE_ON 0x10
+#define DEBUG_GDB_MEM_LOC 0x9E008060
+
+/*
+Save special register designated by 'reg' into
+backup space calculated by offset 'loc * 4' from
+memory pointed by a3.
+*/
+.macro	SAVE_	reg, loc
+	rsr	a1, \reg
+	s32i	a1, a3, \loc * 4
+.endm
+
+.macro	SAVE	reg
+	SAVE_	\reg, \reg
+.endm
+
+/*
+Load special register designated by 'reg' from
+backup space calculated by offset 'loc * 4' from
+memory pointed by a3.
+*/
+.macro	LOAD_	reg, loc
+	l32i	a1, a3, \loc * 4
+	wsr	a1, \reg
+.endm
+
+.macro	LOAD	reg
+	LOAD_	\reg, \reg
+.endm
+
+
+
+.text
+
+/*
+Temporary stack for context switch
+TODO: move it to dedicated GDB_STACK section
+*/
+
+gdb_stack:
+	.fill 0x1000 , 4 , 0
```

Comment:
Just a though, 0x1000 is the default stack size (defined in memory.h) so do we want to use the same macro for size ?

Reply:
Well I haven't made any serious GDB stack size calculation yet. I've just picked some arbitrary stack size just for the sake of having separate stack, that's all. As for using memory.h define, I don't think it is a good idea, I would just make a define in include/gdb/* for it.

Reply:
Also, I wouldn't put too much effort to make this temporary stack here pretty and well documented since I will move  it completely soon...